tests/dags/providers/provider_api_scripts/test_provider_data_ingester.py

```diff
@@ -0,0 +1,181 @@
+import json
+import os
+import unittest
+from unittest.mock import call, patch
+
+from common.storage.audio import AudioStore, MockAudioStore
+from common.storage.image import ImageStore, MockImageStore
+
+from tests.dags.providers.provider_api_scripts.resources.provider_data_ingester.mock_provider_data_ingester import (
+    AUDIO_PROVIDER,
+    EXPECTED_BATCH_DATA,
+    IMAGE_PROVIDER,
+    MockProviderDataIngester,
+)
+
+
+RESOURCES = os.path.join(
+    os.path.abspath(os.path.dirname(__file__)), "resources/provider_data_ingester"
+)
+
+
+class TestProviderDataIngester(unittest.TestCase):
+    def setUp(self):
+        self.ingester = MockProviderDataIngester()
+
+        # Use mock media stores
+        self.audio_store = MockAudioStore(AUDIO_PROVIDER)
+        self.image_store = MockImageStore(IMAGE_PROVIDER)
+        self.ingester.media_stores = {
+            "audio": self.audio_store,
+            "image": self.image_store,
+        }
+
+    def _get_resource_json(self, json_name):
+        with open(os.path.join(RESOURCES, json_name)) as f:
+            resource_json = json.load(f)
+        return resource_json
+
+    def test_init_media_stores(self):
+        ingester = MockProviderDataIngester()
+
+        # We should have two media stores, with the correct types
+        assert len(ingester.media_stores) == 2
+        assert isinstance(ingester.media_stores["audio"], AudioStore)
+        assert isinstance(ingester.media_stores["image"], ImageStore)
+
+    def test_batch_limit_is_capped_to_ingestion_limit(self):
+        with patch(
+            "providers.provider_api_scripts.provider_data_ingester.Variable"
+        ) as MockVariable:
+            MockVariable.get.side_effect = ["200"]
+
+            ingester = MockProviderDataIngester()
+            assert ingester.batch_limit == 200
+
+    def test_get_batch_data(self):
+        response_json = self._get_resource_json("complete_response.json")
+        batch = self.ingester.get_batch_data(response_json)
+
+        assert batch == EXPECTED_BATCH_DATA
+
+    def test_process_batch_adds_items_to_correct_media_stores(self):
+        with (
+            patch.object(self.audio_store, "add_item") as audio_store_mock,
+            patch.object(self.image_store, "add_item") as image_store_mock,
+        ):
+            record_count = self.ingester.process_batch(EXPECTED_BATCH_DATA)
+
+            assert record_count == 3
+            assert audio_store_mock.call_count == 1
+            assert image_store_mock.call_count == 2
+
+    def test_ingest_records(self):
+        with (
+            patch.object(self.ingester, "get_batch") as get_batch_mock,
+            patch.object(
+                self.ingester, "process_batch", return_value=3
+            ) as process_batch_mock,
+            patch.object(self.ingester, "commit_records") as commit_mock,
+        ):
```

Comment:
I'm so glad that you can now have multiple with statements separated by a comma! It's a recent feature as of 3.10[^1] and it make this scenario look so much cleaner 💯 

[^1]: [Changed in version 3.10: Support for using grouping parentheses to break the statement in multiple lines.](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement)